src/vs/workbench/services/positronDataExplorer/browser/positronDataExplorerService.ts

```diff
@@ -190,6 +201,22 @@ class PositronDataExplorerService extends Disposable implements IPositronDataExp
 		this._register(this._runtimeSessionService.onDidChangeRuntimeState(stateEvent => {
 			// console.log(`++++++++++ PositronDataExplorerService: onDidChangeRuntimeState from ${stateEvent.old_state} to ${stateEvent.new_state}`);
 		}));
+
+		// This is a temporary mechanism for extensions like positron-duckdb implementing
+		// a data explorer backend to be able to invoke the frontend methods
+		// (updates, async column profiles) normally invoked by a language runtime kernel
+		this._register(CommandsRegistry.registerCommand('positron-data-explorer.sendUiEvent',
+			(accessor: ServicesAccessor, event: DataExplorerUiEvent) => {
+				const handler = this._uiEventCommandHandlers.get(event.uri);
+
+				// If not event handler registered, ignore for now
+				if (handler === undefined) {
+					return;
+				}
+
+				return handler(event);
+			}
+		));
```

Comment:
This is the command needed to route async column profiles back from the extension into the UI